### PR TITLE
perf: batch Vulkan decode attention into one vkQueueSubmit per layer

### PR DESCRIPTION
**Pull request by ericcurtin**
## What
`kv_ops.py`'s module docstring states its whole design point: "batch all ops for a full transformer layer into one vkQueueSubmit" to avoid "~150µs driver overhead each". `attention.py`'s `_try_vulkan_decode` was the one place still violating that: it called the single-token `paged_attn_decode_f16`/`f32` in a Python `for` loop, once per token in the decode batch, each issuing its own separate `ctx.execute_batch` (one vkQueueSubmit + one fence wait) — so a batch of N concurrent decode requests paid N submits per attention layer per decode step instead of 1.

## Fix
- `kv_ops.py`: `_build_paged_attn_decode_op()` factors the single-token `paged_attn_decode`'s validation + op-tuple construction out of submission, so multiple tokens' ops can be collected and submitted together. `_paged_attn_decode_batch()` (plus public `paged_attn_decode_batch_f16`/`f32` wrappers) builds one op per `(query, block_table, seq_len)` triple in the batch and submits them all via a single `ctx.execute_batch` call. The existing single-token `paged_attn_decode_f16`/`f32` API is unchanged (still used/tested elsewhere) and now implemented as a one-op call through the same shared builder.
- `attention.py`: `_try_vulkan_decode` now calls the batched function once per attention layer instead of looping over the single-token one.

## Measured impact
On this hardware (`kv_ops.py`'s own decode primitives, 8 query heads, head_size=64):

```
batch_size=  1  OLD:   0.272 ms  NEW:   0.266 ms  speedup: 1.02x
batch_size=  4  OLD:   1.121 ms  NEW:   0.765 ms  speedup: 1.47x
batch_size=  8  OLD:   2.668 ms  NEW:   1.717 ms  speedup: 1.55x
batch_size= 32  OLD:  16.487 ms  NEW:  12.785 ms  speedup: 1.29x
batch_size= 64  OLD:  37.223 ms  NEW:  20.051 ms  speedup: 1.86x
```

The win grows with batch size, as expected for amortizing per-submit overhead across more work; `batch_size=1` correctly shows ~1.0x, since there's nothing to batch.

## Testing
Added to `tests/python/test_kv_ops.py`:
- `test_paged_attn_decode_batch_matches_per_token_calls_and_torch_reference`: batched output matches both the per-token calls it replaces AND an independent pure-PyTorch attention reference, for a batch of **two sequences with different seq_lens/block_tables/K-V data** sharing one cache — not just a same-shape trivial case.
- `test_paged_attn_decode_batch_validates_mismatched_list_lengths`
- `test_paged_attn_decode_batch_empty_returns_empty_list`

`ruff`/`mypy` clean; all 36 Python tests pass (2 skipped, no vllm installed in this environment), including all pre-existing `kv_ops`/`attention` tests unaffected; no Rust changes, so all 19 lib tests continue to pass unaffected.

### CODE REVIEW

File: vllm_vulkan/attention.py [hunk 1]
```diff
@@ -214,36 +214,38 @@ def _try_vulkan_decode(
...
+            # be the one place still issuing a separate submit+fence-wait
+            # per token, once per attention layer per decode step.
+            outs = decode_batch(
+                ctx,
+                entry.layout,
+                entry.cache,
+                _PER_LAYER_KV_CACHE_INDEX,
+                [query[i].detach().to("cpu") for i in range(num_actual_tokens)],
+                [block_table[i] for i in range(num_actual_tokens)],
+                [int(seq_lens[i]) for i in range(num_actual_tokens)],
```
[the remark this answers is not in the document]
Comment (ericcurtin) [reply vllm_vulkan/attention.py]: Fixed in d73d5e0 — now uses `query[:num_actual_tokens].detach().to('cpu').unbind(0)`, `block_table.unbind(0)`, and `seq_lens.tolist()` exactly as suggested. Verified equivalence with the old per-index approach via a standalone smoke test.

File: vllm_vulkan/kv_ops.py [hunk 2]
```diff
@@ -274,24 +341,104 @@ def _paged_attn_decode(
...
+            q=q,
+            block_table=block_table,
+            seq_len=seq_len,
+            scale=scale,
+            shader_name=shader_name,
+            coop_shader_name=coop_shader_name,
+            dtype_size=dtype_size,
+        )
+        ops.append(op)
+        shapes.append((num_q_heads, head_size))
```
[the remark this answers is not in the document]
Comment (ericcurtin) [reply vllm_vulkan/kv_ops.py:479]: Fixed in d73d5e0 — added _resolve_paged_attn_decode_dispatch() which resolves shader selection (_select_decode_shader, the FFI call) plus cache/dtype validation exactly once per batch, and refactored _build_paged_attn_decode_op() to accept the pre-resolved dispatch_shader_name/spec instead of re-deriving them per token. Re-measured after this fix: speedups improved further (e.g. batch_size=8 went from 1.55x to 1.73x).
